logstash-core/src/main/java/org/logstash/common/Util.java

```diff
@@ -0,0 +1,31 @@
+package org.logstash.common;
+
+import java.security.MessageDigest;
+
+/**
+ * Created by andrewvc on 12/23/16.
+ */
+public class Util {
+    // Modified from http://stackoverflow.com/a/11009612/11105
+    public static String sha256(String base) {
+        try{
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] hash = digest.digest(base.getBytes("UTF-8"));
+            return bytesToHexString(hash);
+        } catch(Exception ex){
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public static String bytesToHexString(byte[] bytes) {
```

Comment:
Hmmm, that's awesome, but I don't know if I want to tie this to the java XML processing functions.